plugins/sla_check.py

```diff
@@ -1,179 +1,112 @@
-    DagRun.dag_id == dag_name,
-    DagRun.execution_date == dt
-  ).first()
-
-  return last_run and last_run.state != State.RUNNING
-
-# This takes in a String formatted \d+hr\d+m\d+s and creates a parsed out
-# timedelta object that resembles the time from midnight the DAG needs to
-# finish by in order to be compliant.
-def parse_timedelta(time):
-  parts = regex.match(time)
-  if not parts:
-    return
-  parts = parts.groupdict()
-  time_params = {}
-  for (name, param) in parts.iteritems():
-    if param:
-      time_params[name] = int(param)
-  return timedelta(**time_params)
-
-# Wraps the above functions into a single call to find only the SLA's which
-# are in violation of their configured SLA.
-def find_violations():
-  slas = find_slas()
-  return filter(lambda s: not check_compliance(s[0], s[1]), slas)
-
 class SLACheckView(BaseView):
   # Diagnostic endpoint which lists out the discovered SLA's and
   # what time they are due at today.
   @expose('/')
   def index(self):
     logging.info("SLA Check index() called")
-    slas = find_slas()
-    slas_with_due = [(x[0], calculate_sla_due(x[1])) for x in slas]
-    return self.render("sla_check/index.html", slas=slas_with_due)
+    slas = Variable.get(SLA_CONFIG_VARIABLE, default_var=dict(), deserialize_json=True)
+    print slas
+    return self.render("sla_check/index.html", slas=slas)
 
   # Wraps the violations for DataDog's format, it will return 200 if
   # there are no violations and 500 and a comma-separated string of
   # dag_id's that are currently in violation.
-  @expose('/healthcheck')
-  def health_check(self):
-    logging.info("SLA Health Check index() called")
+  @expose('/slacheck')
+  def sla_check(self):
+    logging.info("SLA Check() called")
 
-    violations = find_violations()
-    violation_names = [x[0] for x in violations]
+    violations = []
 
-    if len(violation_names) > 0:
-      return ', '.join(violation_names), 500
+    # Load SLA's and DAG configurations
+    slas = Variable.get(SLA_CONFIG_VARIABLE, default_var=dict(), deserialize_json=True)
+    dagbag = DagBag()
 
-    else:
-      return '', 200
+    # Filter down to DAG's with SLA's
+    enforceable_slas = [x for x in slas if x['dag_id'] in dagbag.dags.keys()]
+
+    for sla in enforceable_slas:
+      dag = dagbag.dags[sla['dag_id']]
+      # Check if this DAG uses an Airflow Cron Preset
+      if dag.schedule_interval in cron_presets:
+        delta = cron_presets[dag.schedule_interval]
+      else:
+        delta = dag.schedule_interval
 
-  # Same data as the DataDog endpoint but wrapped in nicer JSON with
-  # additional metadata. This endpoint is mainly useful for potential
-  # future data collecting.
-  @expose('/health')
-  def health_status(self):
-    logging.info("SLA Health Check index() called")
-    base_response = get_base_response()
+      # Compute when it should have run last
+      runs = date_range(start_date=dag.start_date, end_date=dag.end_date, delta=delta)
```

Comment:
There's a method on dag that automatically computes all run dates in a range: `dag.date_range(start_date, end_date=end_date)`